cfme/generic_objects/definition/ui.py

```diff
@@ -112,12 +112,22 @@ def generic_objects(self):
     return self.collections.generic_objects
 
 
+@MiqImplementationContext.external_for(GenericObjectDefinition.generic_object_buttons.getter, ViaUI)
+def generic_object_buttons(self):
+    return self.collections.generic_object_buttons
```

Comment:
This is consistent with the property above, but I think you should take a hard look at the nuance here - this property, for ViaUI, is actually just returning something attached to the model of the generic object defintion - its not actually returning what generic object buttons are attached to the generic object definition _as shown by the CFME UI_.

Reply:
Yeah I thought it was weird to have it defined in this way, but wanted to remain consistent with existing framework. I will probably change in the near future!